dev/AnimatedVisualPlayer/AnimatedVisualPlayer.cpp

```diff
@@ -342,10 +343,14 @@ AnimatedVisualPlayer::AnimatedVisualPlayer()
 AnimatedVisualPlayer::~AnimatedVisualPlayer()
 {
     // Ensure any outstanding play is stopped.
-    // NOTE: Stop() can cause reentrance when clients react to DP changes, but because
-    //       we're in the destructor we know that there aren't any clients who can reach
-    //       us, so reentrance is not a concern. 
-    Stop();
+    if (m_nowPlaying)
+    {
+        // To stop and destroy m_nowPlaying we need to call Complete()
+        // We need to detach m_nowPlaying first so that Complete()
+        // will not try to call us (owner), since we already in the destructor
+        auto nowPlaying = std::move(m_nowPlaying);
+        nowPlaying->Complete();
+    }
```

Comment:
Can whatever cleanup needs to happen just be in the destructor of this object?

Reply:
As I see `AnimatedVisualPlayer::PlayAsync` [(code)](https://github.com/microsoft/microsoft-ui-xaml/pull/4955/commits/a2dadc322c90ea5a7b46bbc923357f051fed5ba8#L642) holds reference to this object too, and it will not release it until we call `CompleteAwaits()`, because of `co_await *thisPlay` [(code)](https://github.com/microsoft/microsoft-ui-xaml/pull/4955/commits/a2dadc322c90ea5a7b46bbc923357f051fed5ba8#L660), is it right?

Reply:
If we think that AnimationPlay outlives AnimatedVisualPlayer then AnimationPlay should reference AnimatedVisualPlayer with winrt::weak_ref and check it when it uses it. 

The comment should also be updated to be more precise -- it won't call the owner but it will check the owner's m_nowPlaying field.

Reply:
I think that since AnimationPlay is declared inside AnimatedVisualPlayer, I can't use weak_ref here?

`an undefined class is not allowed as an argument to compiler intrinsic type trait '__is_base_of'`

Reply:
My other suggestion of nulling m_owner in AnimatedVisualPlayer destructor is sufficient. Then the pointer won't be dangling after destruction.